src/llm/generation_chains/rooms_chain.py

```diff
@@ -0,0 +1,130 @@
+from marshmallow import ValidationError
+import re
+from typing import Optional
+import yaml
+
+from llm.marshmallow.schemas import RoomSchema
+
+class RoomYamlOutputParser(BaseOutputParser):
+    """Parse the output of an LLM call to YAML."""
+
+    def parse(self, text: str) -> Optional[RoomSchema]:
+        """Parse the output of an LLM call."""
+        match = (
+            re.search(r"- [rR]oom:[\s\S]*", text)
+            or re.search(r"[rR]oom:[\s\S]*", text)
+            or re.search(r"- [rR]]oom:[\s\S]*\n", text)
+            or re.search(r"[rR]oom:[\s\S]*\n", text)
+        )
+        group = match.group(0)
+        
+        if group.startswith('- '):
+            group = group[2:]
+        
+        if '`' in group:
+            group = re.search(r'([^`]+)`', group).group(1).strip()
+        
+        obj = yaml.safe_load(group)
+        top_level_key = list(obj.keys())[0]
+        try:
+            return RoomSchema().load(obj[top_level_key])
+        except ValidationError as err:
+            print(err.messages)
+            return None
+
+class RoomsChain:
+    def __init__(self, llm, rooms_layout):
+        self._llm = llm
+
+        self._rooms_layout = rooms_layout
+        self._rows, self._columns = len(rooms_layout), len(rooms_layout[0])
+
+        self._suspect_prompt = PromptTemplate.from_template(
+            """
+            <s>[INST] <<SYS>>
+            You are a crime storyteller.
+            <<SYS>>
+
+            Describe a room given a theme and information about a suspect person in that room.
+            Room is described by its name and a description.
+            An example is below. Note that you must output only created description converted to YAML where all properties are on same level.
+            
+            Example answer:
+            room:
+              name: "The Ancient Manuscript Chamber"
+              description: "Ancient Manuscript Chamber is known as the Preservation Gallery. It serves as a critical space for the maintenance and restoration of the library's valuable scrolls and manuscripts. The room is well-lit with natural sunlight streaming in through high windows, revealing a contrast to the dimly lit chamber next door. Large, polished wooden tables occupy the center of the room, where scholars and assistants like Lucius meticulously repair and transcribe ancient texts. Rows of neatly organized shelves hold scrolls in protective containers. The air in this room carries the faint aroma of aged parchment and the careful attention of those dedicated to preserving the knowledge of the ages. Lucius, the Librarian's Assistant, was seen in this room working diligently during the time of Drusilla's murder, and he claims he had no involvement in the crime." [/INST]
+            
+            Give the output for the following theme: {theme} and suspect: {entity}
+            """
+        )
+
+        self._victim_prompt = PromptTemplate.from_template(
```

Comment:
Prečo sa to volá victim prompt keď je tam roomka

Reply:
pomenoval som to podla toho, ze ten prompt je pre room, kde je victim. ta druha je pre room, kde je suspect. ale mas pravdu trochu, pozmenim to.

Reply:
jasne, dik 🙂